components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1420,4 +1427,29 @@ private void setLoggedInOrgIdInRequest(AuthenticatedIdPData authenticatedIdPData
             }
         }
     }
+
+    private void setUserOrganization(AuthenticatedUser authenticatedUser, AuthenticatorConfig authenticatorConfig) {
+
+        String userOrganization = null;
+        if (authenticatedUser.getUserAttributes() != null && isLoggedInWithOrganizationLogin(authenticatorConfig)) {
+            for (Map.Entry<ClaimMapping, String> userAttributes : authenticatedUser.getUserAttributes().entrySet()) {
+                if (FrameworkConstants.USER_ORGANIZATION_CLAIM.equals(
+                        userAttributes.getKey().getLocalClaim().getClaimUri())) {
+                    userOrganization = userAttributes.getValue();
+                }
+            }
+        }
+
+        if (userOrganization == null) {
+            try {
+                userOrganization = FrameworkServiceDataHolder.getInstance().getOrganizationManager()
+                        .resolveOrganizationId(authenticatedUser.getTenantDomain());
+            } catch (OrganizationManagementClientException e) {
+                LOG.debug("Organization ID is not found for the given tenant domain");
+            } catch (OrganizationManagementException e) {
```

Comment:
Why we need to catch these seperately?

Reply:
The client exception is thrown when we request for organization ID for a tenant who doesn't have an org-id (b2b disabled tenants). Other exception is to handle the server error if occured.

Reply:
The above implementation is now obsolete